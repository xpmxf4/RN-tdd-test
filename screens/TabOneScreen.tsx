import * as React from "react";
import { StyleSheet, ScrollView } from "react-native";
import { gql, useQuery } from "@apollo/client";

import { Text, View } from "../components/Themed";

interface name {
  name: String;
}

export const GET_DOG_QUERY = gql`
  query GetDog($name: String) {
    dog(breed: $name) {
      id
      name
      breed
    }
  }
`;

export default function TabOneScreen({ name }: name) {
  const { loading, error, data } = useQuery(GET_DOG_QUERY, {
    variables: { name },
  });

  return (
    <View style={styles.container}>
      <ScrollView>
        {loading ? (
          <Text>loading....</Text>
        ) : error ? (
          <Text>{error.message}</Text>
        ) : (
          data?.dogs?.map((dogs: any) => <Text key={dogs.id}>{dogs.breed}</Text>)
        )}
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "pink",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
});
