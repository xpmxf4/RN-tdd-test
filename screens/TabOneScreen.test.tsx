import * as React from "react";
import { ScrollView } from "react-native";
import TestRenderer from "react-test-renderer";
import waitForExpect from "wait-for-expect";
import { MockedProvider } from "@apollo/client/testing";
import TabOneScreen, { GET_DOG_QUERY } from "./TabOneScreen";
import { waitFor } from "@testing-library/dom";
import { Text } from "../components/Themed";

const mocks = [
  {
    request: {
      query: GET_DOG_QUERY,
      variables: {
        name: "Buck",
      },
    },
    result: {
      data: {
        dog: { id: "1", name: "Buck", breed: "poodle" },
      },
    },
  },
];

test("it renders without error", () => {
  const component = TestRenderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <TabOneScreen name={"poodle"} />
    </MockedProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render dog", () => {
  const dogMock = {
    request: {
      query: GET_DOG_QUERY,
      vaiables: { name: "Buck" },
    },
    result: {
      data: { dog: { id: 1, name: "Buck", breed: "poodle" } },
    },
  };

  const component = TestRenderer.create(
    <MockedProvider mocks={[dogMock]} addTypename={false}>
      <TabOneScreen name={"Buck"} />
    </MockedProvider>
  );

  const scrollView = component.root.findAllByType(Text);

  // await new Promise((resolve) => setTimeout(resolve, 0));
  expect(scrollView.join("").).toContain("loading...");
});
