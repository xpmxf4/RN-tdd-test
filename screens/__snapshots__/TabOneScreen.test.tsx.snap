// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders without error 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "pink",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            undefined,
          ]
        }
      >
        loading....
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
